Cabal/Distribution/GetOpt.hs

```diff
@@ -234,6 +187,7 @@ shortOpt y ys rs optDescr = short ads ys rs
         short (OptArg f _:_) xs rest     = (Opt (f (Just xs)),rest)
         short []             [] rest     = (UnreqOpt optStr,rest)
         short []             xs rest     = (UnreqOpt (optStr++xs),rest)
+        -- this part is different vs upstream GetOpt ^^
```

Comment:
Pasting the diff between ghc's and cabal's difference in fmtLong:

-        short []             xs rest     = (UnreqOpt (optStr++xs),rest)
+        short []             xs rest     = (UnreqOpt optStr,('-':xs):rest)